tests/integration/test_dictionaries_wait_for_load/test.py

```diff
@@ -0,0 +1,41 @@
+import pytest
+from helpers.cluster import ClickHouseCluster
+from helpers.test_tools import TSV
+
+
+DICTIONARY_FILES = [
+    "dictionaries/long_loading_dictionary.xml",
+]
+
+cluster = ClickHouseCluster(__file__)
+node1 = cluster.add_instance(
+    "node1",
+    main_configs=["configs/wait_for_dictionaries_load.xml"],
+    dictionaries=DICTIONARY_FILES,
+)
+
+node0 = cluster.add_instance("node0", dictionaries=DICTIONARY_FILES)
+
+
+@pytest.fixture(scope="module", autouse=True)
+def started_cluster():
+    try:
+        cluster.start()
+        yield cluster
+    finally:
+        cluster.shutdown()
+
+
+def get_status(instance, dictionary_name):
+    return instance.query(
+        "SELECT status FROM system.dictionaries WHERE name='" + dictionary_name + "'"
+    ).rstrip("\n")
+
+
+def test_wait_for_dictionaries_load():
+    assert get_status(node0, "long_loading_dictionary") == "NOT_LOADED"
```

Comment:
So, we expect that `get_status` is called immediately after node created. May be flaky, especially with sanitizer builds, but probably ok, let's see

Reply:
I don't think it's going to be flaky. But if it is so we'll fix it then.